apiserver/root.go

```diff
@@ -740,9 +755,14 @@ func (ctx *facadeContext) ControllerDB() (changestream.WatchableDB, error) {
 	return db, errors.Trace(err)
 }
 
-// DBDeleter returns a database deleter.
-func (ctx *facadeContext) DBDeleter() coredatabase.DBDeleter {
-	return ctx.r.shared.dbDeleter
+// ServiceFactory returns the services factory for the current model.
+func (ctx *facadeContext) ServiceFactory() servicefactory.ServiceFactory {
+	return ctx.r.serviceFactory
+}
+
+// ServiceFactoryForModel returns the services factory for the given model.
+func (ctx *facadeContext) ServiceFactoryForModel(uuid string) servicefactory.ServiceFactory {
```

Comment:
I don't like having these 2 methods.

If we look at `serveConn`, it decides there whether we have a model UUID for the request (socket lifetime in our case). If there is none we use the controller's model ID.

I think we need to do something similar, but it will rest on a decision not-yet-made, which is do the controller _model_ concerns live in the same or a different database than `controller`.

Reply:
I removed this, as we discussed, let's not design CMR until we need to.